src/node/desktop/src/main/menu-callback.ts

```diff
@@ -485,16 +488,25 @@ export class MenuCallback extends EventEmitter {
    * @param enabled Whether the main menu bar should be enabled or disabled (replaced with a placeholder).
    */
   setMainMenuEnabled(enabled: boolean) {
-    const stubWithPlaceholder = !enabled && !this.savedMenu;
-    if (stubWithPlaceholder) {
-      // Save a copy of the current menu bar so we can restore it later
+    if (!enabled && !this.savedMenu) {
       this.savedMenu = Menu.getApplicationMenu();
-      this.showPlaceholderMenu();
+      if (this.savedMenu) {
+        const disabledMenu = Menu.buildFromTemplate(this.recursiveCopy(this.mainMenuTemplate));
+        disabledMenu?.items?.forEach((item) => {
+          item.submenu?.items.forEach((subItem) => {
+            // keep some commands enabled
+            subItem.enabled = ['cut', 'copy', 'paste', 'redo', 'hide', 'hideOthers', 'unhide', 'selectAll'].includes(subItem.role ?? '');
+          });
+        });
+        Menu.setApplicationMenu(disabledMenu);
+        this.mainMenu = disabledMenu;
+      }
```

Comment:
IIUC, this means we can remove `showPlaceholderMenu()` altogether?

Reply:
Oh, that's something that you had added before to handle the modals. I'll remove it!

Reply:
And I'm already mixing myself up. It looks like it was refactored for that modal change. Gary had initially added the placeholder and I'm pretty sure it's so that the top-level menus show up during startup until GWT sends the menu items.

Reply:
ah right, the menu items aren't yet ready during startup, so we need the placeholder 👍